### PR TITLE
[Event] + change the event on flush to on pre flush for element file into RevisionListener

### PR DESCRIPTION
**Pull request by fkroockmann**
Bad use of event cause of change the data into a flush event instead of a pre flush event.

### CODE REVIEW

File: Event/Listener/RevisionListener.php [hunk 1]
```diff
@@ -120,10 +120,15 @@ public static function onFlushElementFile(Event $event)
 
         $revision = $event->getTarget();
         $content = $revision->getContent();
-
+        
+        if ($uow->isScheduledForDelete($content)) {
+            return;
```
Comment (eric-chau) [inline Event/Listener/RevisionListener.php]: You have to check if there is a file that exists at its ``path`` property location cause here it's a hard delete of the entity (not a simple change of its ``state``).

Comment (fkroockmann) [inline Event/Listener/RevisionListener.php]: What do you think @crouillon, we delete the file or we keep it ?

**Comment by fkroockmann** [comment]
@eric-chau, i create a issue for add a delete treatment.
https://github.com/backbee/BackBee/issues/312

**Comment by eric-chau** [comment]
ok @fkroockmann, atleast we won't forget to do it later. Thanks.

**Pull request merged**